### PR TITLE
[20.09] trigger: 0.6.6 -> 0.6.6.1

### PR DESCRIPTION
**Pull request by p-h**
(cherry picked from commit f85e2e1eeb442c0b0a8afb8e224f8a6a039b5664)

###### Motivation for this change
backport of: #100449 
ZHF: #97479

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/trigger/default.nix [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 
   src = fetchurl {
     url = "mirror://sourceforge/trigger-rally/${name}.tar.gz";
-    sha256 = "08qa2f2s8zyn42ff6jb1gsi64d916020ixkzvl16kbb88rabqra8";
+    sha256 = "016bc2hczqscfmngacim870hjcsmwl8r3aq8x03vpf22s49nw23z";
   };
 
-  buildInputs = [ SDL2 freealut SDL2_image openal physfs zlib libGLU libGL glew ];
+  buildInputs =
+    [ SDL2 freealut SDL2_image openal physfs zlib libGLU libGL glew tinyxml-2 ];
```
Comment (jonringer) [inline pkgs/games/trigger/default.nix:13]:
```suggestion
  buildInputs = [
    SDL2
    freealut
    SDL2_image
    openal
    physfs
    zlib
    libGLU
    libGL
    glew
    tinyxml-2
  ];
```

File: pkgs/games/trigger/default.nix [hunk 2]
```diff
@@ -1,20 +1,23 @@
-{ fetchurl, stdenv, runtimeShell
-, SDL2, freealut, SDL2_image, openal, physfs, zlib, libGLU, libGL, glew }:
+{ fetchurl, stdenv, runtimeShell, SDL2, freealut, SDL2_image, openal, physfs
+, zlib, libGLU, libGL, glew, tinyxml-2 }:
 
 stdenv.mkDerivation rec {
-  name = "trigger-rally-0.6.6";
+  name = "trigger-rally-0.6.6.1";
```
Comment (jonringer) [inline pkgs/games/trigger/default.nix:5]:
```suggestion
  pname = "trigger-rally";
  version = "0.6.6.1";
```

File: pkgs/games/trigger/default.nix [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
 
   src = fetchurl {
     url = "mirror://sourceforge/trigger-rally/${name}.tar.gz";
```
Comment (jonringer) [inline pkgs/games/trigger/default.nix:8]:
```suggestion
    url = "mirror://sourceforge/trigger-rally/${pname}-${version}.tar.gz";
```

**Comment by p-h** [comment]
I applied your suggestions. These are not in master right now. Should I open another pr for that?

**Comment by jonringer** [comment]
> I applied your suggestions. These are not in master right now. Should I open another pr for that?

Apparently I forgot to hit submit.

I know that a lot of these changes are already in master, please make a new PR against master with the changes and cherry-pick those two commits into this PR please.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 100492 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>trigger</li>  </ul></details>

**Pull request merged**